### PR TITLE
added itunes container & tests

### PR DESCRIPTION
**Pull request by adidoshiws**
### Ticket Link

---

### Related Links

---

### Description

- Added itunesContainer with react-floki
- API call in the change handler
- Updated services for itunes
- Tests for reducers, saga, selectors, index.js & services

### Steps to Reproduce / Test

---

---

### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's
containerPRDemo0

---

### Live Link

---

### CODE REVIEW

File: app/containers/ItunesContainer/index.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  useEffect(() => {
+    if (trackName && !tracksData?.results?.length) {
+      dispatchItuneTracks(trackName);
+      setLoading(true);
+    }
+  }, []);
+
+  const handleOnChange = (tName) => {
+    if (!isEmpty(tName)) {
+      dispatchItuneTracks(tName);
```
Comment (ritwik-wednesday) [inline app/containers/ItunesContainer/index.js:34]: can we handle this in the useEffect.

File: app/containers/ItunesContainer/index.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      setLoading(true);
+    } else {
+      dispatchClearItuneTracks();
```
Comment (ritwik-wednesday) [inline app/containers/ItunesContainer/index.js:37]: can we handle this in the useEffect.

File: app/containers/ItunesContainer/saga.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { getTracks } from '@services/itunesApi';
+import { put, call, takeLatest } from 'redux-saga/effects';
+import itunesContainerTypes, { itunesContainerCreators } from './reducer';
+
+const { REQUEST_GET_ITUNE_TRACKS } = itunesContainerTypes;
+
+const { successGetItuneTracks, failureGetItuneTracks } = itunesContainerCreators;
+
+export function* getItuneTracks(action) {
+  const response = yield call(getTracks, action.trackName);
```
Comment (ritwik-wednesday) [inline app/containers/ItunesContainer/saga.js:10]: lets use `try/catch`

Comment (adidoshiws) [reply app/containers/ItunesContainer/saga.js:10]: Yes, added try catch!

File: app/containers/ItunesContainer/index.js (hunk 2 above)
Comment (adidoshiws) [reply app/containers/ItunesContainer/index.js:37]: Sure, made the changes.

File: app/containers/ItunesContainer/index.js (hunk 1 above)
Comment (adidoshiws) [reply app/containers/ItunesContainer/index.js:34]: Sure, made the changes.

Comment (ritwik-wednesday) [reply app/containers/ItunesContainer/index.js:34]: Instead on calling the dispatch func here, we can use a state and change it which would trigger useEffect to make the dispatch call.

File: app/containers/ItunesContainer/index.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    let trackError;
+    if (isEmpty(trackName)) {
+      trackError = 'track_search_default';
+    }
+    return !loading && trackError && <></>;
+  };
+
+  return (
+    <>
+      <input
```
Comment (tousifws) [inline app/containers/ItunesContainer/index.js:53]: Use antD component https://ant.design/components/input/

File: app/containers/ItunesContainer/tests/index.test.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  });
+
+  // it('should call dispatchItuneTracks on submit', async () => {
+  //   const trackName = 'rockstar';
+  //   const { getByTestId } = renderProvider(<ItunesContainer dispatchItuneTracks={submitSpy} />);
+  //   fireEvent.keyDown(getByTestId('search-bar'), { keyCode: 13, target: { value: trackName } });
+
+  //   await timeout(500);
+  //   expect(submitSpy).toBeCalledWith(trackName);
+  // });
```
Comment (tousifws) [inline app/containers/ItunesContainer/tests/index.test.js]: If not needed please remove it

**Review by tousifws — COMMENTED** [review]
Please attach a gif or video showing your current status of your app

File: app/containers/ItunesContainer/tests/index.test.js (hunk 5 above)
Comment (adidoshiws) [reply app/containers/ItunesContainer/tests/index.test.js]: Sure removing it.

File: app/containers/ItunesContainer/index.js (hunk 4 above)
Comment (adidoshiws) [reply app/containers/ItunesContainer/index.js:53]: Sure, added now.

File: app/containers/ItunesContainer/index.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    if (!isEmpty(tName)) {
+      dispatchItuneTracks(tName);
+      setLoading(true);
+    } else {
+      dispatchClearItuneTracks();
+    }
+  };
+
+  useEffect(() => {
+    handleOnChange();
```
Comment (ritwik-wednesday) [inline app/containers/ItunesContainer/index.js]: do we need this?

File: app/containers/ItunesContainer/index.js (hunk 1 above)
Comment (ritwik-wednesday) [reply app/containers/ItunesContainer/index.js:34]: can you try to do it in useEffect

File: app/containers/ItunesContainer/index.js (hunk 6 above)
Comment (adidoshiws) [reply app/containers/ItunesContainer/index.js]: No, that was unnecessary, removed it.

**Review by ritwik-wednesday — APPROVED** [review]

**Pull request merged**